app/src/main/java/com/parishod/watomatic/NotificationService.java

```diff
@@ -149,18 +151,55 @@ private boolean canSendReplyNow(StatusBarNotification sbn){
     }
 
     private void logReply(StatusBarNotification sbn){
-        String title = sbn.getNotification().extras.getString("android.title");
         messageLogsDB = MessageLogsDB.getInstance(getApplicationContext());
         int packageIndex = messageLogsDB.appPackageDao().getPackageIndex(sbn.getPackageName());
         if(packageIndex <= 0){
             AppPackage appPackage = new AppPackage(sbn.getPackageName());
             messageLogsDB.appPackageDao().insertAppPackage(appPackage);
             packageIndex = messageLogsDB.appPackageDao().getPackageIndex(sbn.getPackageName());
         }
-        MessageLog logs = new MessageLog(packageIndex, title, sbn.getNotification().when, customRepliesData.getTextToSendOrElse(null), System.currentTimeMillis());
+        MessageLog logs = new MessageLog(packageIndex, getTitle(sbn), sbn.getNotification().when, customRepliesData.getTextToSendOrElse(null), System.currentTimeMillis());
         messageLogsDB.logsDao().logReply(logs);
     }
 
+    private String getTitle(StatusBarNotification sbn) {
+        String title = "";
+        if(sbn.getNotification().extras.getBoolean("android.isGroupConversation")) {
+            title = sbn.getNotification().extras.getString("android.hiddenConversationTitle");
+            //Just to avoid null cases, if by any chance hiddenConversationTitle comes null for group message
+            // then extract group name from title
+            if(title == null) {
+                title = sbn.getNotification().extras.getString("android.title");
+                int index = title.indexOf(':');
+                if (index != -1) {
+                    title = title.substring(0, index);
+                }
+            }
+
+            //To eliminate the case where group title has number of messages count in it
+            Parcelable b[] = (Parcelable[]) sbn.getNotification().extras.get(Notification.EXTRA_MESSAGES);
+            if(b != null && b.length > 1){
+                int startIndex = title.lastIndexOf('(');
+                int endIndex = title.lastIndexOf(')');
+                if (startIndex != -1 && endIndex != -1 && endIndex > startIndex) {
+                    String strBetween = title.substring(startIndex + 1, endIndex);
+                    //Split the string and check if it has 2 components and 1st component is integer
+                    String[] splitStr = strBetween.split(" ");
+                    try {
+                        if(splitStr.length > 1 && Integer.parseInt(splitStr[0]) != -1){
```

Comment:
Integer.parseInt() does not work with non ascii digits. I am not sure how WhatsApp or others show for other languages. Will check that.
Also, do we even have to check that? If `b.length > 1`, is it not guarateed to have `(<num> <messages>)`. If yes, can we just remove last `(*)`?

Reply:
Im checking length to avoid removing any (*) present in group name

Reply:
Checking for length is good but asking if parseInt check can be removed:
`if(splitStr.length > 1 && Integer.parseInt(splitStr[0]) != -1)` because we are already checking the length so is it safe to assume (*) exists?

Only concern with isInteger check is language compatibility

Reply:
Yeah have to see if it works for other language but group name can also have two words inside brackets so kept integer check